### PR TITLE
complete intercom

### PR DESCRIPTION
**Pull request by mbliakharskyi**
## Description

Please include a summary of the changes and issues addressed in this PR. Also, explain the motivation and context of each update. List any dependencies or prerequisites needed for implementing this change.

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/intercom/src/connectors/users.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+const intercomResponseSchema = z.object({
+  pages: z.object({
+    page: z.number().nullable(),
+    per_page: z.number().nullable(),
+    next: z.object({
+      starting_after: z.string().nullable(),
+    }).optional(),
+  }).optional(),
+  admins: z.array(intercomUserSchema),
```
Comment (elba-integration) [inline apps/intercom/src/connectors/users.ts:20]:
```suggestion
  admins: z.array(z.unknown()),
```
The point is to let unexpected admin data pass the `.parse(data)` validation, then separate the invalid admins from the the valid one.

File: apps/intercom/src/inngest/middlewares/rate-limit-middleware.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      onFunctionRun: ({ fn }) => {
+
+        return {
+          transformOutput: (ctx) => {
+            const {
+              result: { error, ...result },
+              ...context
+            } = ctx;
+            const retryAfter =
+              error instanceof IntercomError && error.response?.headers.get('Retry-After');
```
Comment (elbalexandre) [inline apps/intercom/src/inngest/middlewares/rate-limit-middleware.ts:31]:
```suggestion
              error instanceof IntercomError && error.response?.headers.get('X-RateLimit-Reset');
```

File: apps/intercom/src/inngest/middlewares/rate-limit-middleware.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+              return;
+            }
+
+            return {
+              ...context,
+              result: {
+                ...result,
+                error: new RetryAfterError(
+                  `Intercom rate limit reached by '${fn.name}'`,
+                  retryAfter,
```
Comment (elbalexandre) [inline apps/intercom/src/inngest/middlewares/rate-limit-middleware.ts:43]:
```suggestion
                  new Date(retryAfter * 1000),
```

File: apps/intercom/src/inngest/functions/users/schedule-users-synchronize.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      await step.sendEvent(
+        'synchronize-users',
+        organisations.map(({ id, region }) => ({
+          name: 'intercom/users.sync.requested',
+          data: {
+            isFirstSync: false,
+            organisationId: id,
+            syncStartedAt: Date.now(),
+            page: null,
+            region
```
Comment (elbalexandre) [inline apps/intercom/src/inngest/functions/users/schedule-users-synchronize.ts:27]: Let's drop the region from event & retrieve it from db when needed. I need to update the template.

Comment (mbliakharskyi) [reply apps/intercom/src/inngest/functions/users/schedule-users-synchronize.ts:27]: Got it.

**Review by elbalexandre — APPROVED** [review]

**Pull request merged**